### ISSUE TITLE
Error. Too many tokens required or invalid API key.

### ISSUE THREAD

**Issue by fuyun-cas** [ac-rad/xdl-generation#12]
Hi, thanks for sharing, very impressive work. While I use it, the execution log shows "Error. Too many tokens required or invalid API key." I am sure the API key works. I use the XDL-description.txt as the description of prompt, the txt in Chem-RnD as instructions. It seems that the XDL-description file plus instructions are more than 1000 tokens. So could you please tell me how to solve the problem?

Execution log:
Model: text-davinci-003
Convert to XDL: To the substrate in DMF added the electrophile and then the KOH, stirred overnight at room temperature. Standard aqueous work up using ether for extraction. Purification by silica-gel chromatography using gradient elution (hexane:ether, 50:1, 20:1, 10:1, 5:1, 2:1) gave the product as a white solid (51%)
Error. Too many tokens required or invalid API key.
XDL: The correct XDL could not be generated.
Final syntax valid: False

**Comment by n-yoshikawa** [comment]
Hello fuyun-cas, I am sorry for the delayed response.

I tested your input with the [Google Colab](https://colab.research.google.com/github/ac-rad/xdl-generation/blob/master/tutorial.ipynb) using my API key, but I could not reproduce the reported problem.

Can you test with the Google Colab with your API key and check if it reproduces?
If it does not reproduce, it could be an installation problem. Can you describe the details of your installation process?

PS: text-davinci-003 failed to generate a correct XDL for your input, but GPT-4 worked.
```
<XDL>
  <Synthesis>  
      <Hardware>
          <Component id='DMF_vessel'/>
          <Component id='ether_vessel'/>
          <Component id='product_vessel'/>
      </Hardware>
  
      <Reagents>
          <Reagent name='substrate'/>
          <Reagent name='electrophile'/>
          <Reagent name='KOH'/>
          <Reagent name='DMF'/>
          <Reagent name='ether'/>
          <Reagent name='water'/>
          <Reagent name='silica gel'/>
          <Reagent name='hexane'/>
      </Reagents>
  
      <Procedure>
          <Add vessel='DMF_vessel' reagent='substrate'/>
          <Add vessel='DMF_vessel' reagent='electrophile'/>
          <Add vessel='DMF_vessel' reagent='KOH'/>
          <StartStir vessel='DMF_vessel'/>
          <Stir vessel='DMF_vessel' time='1440'/> <!-- overnight in minutes -->
          <StopStir vessel='DMF_vessel'/>
          <Add vessel='ether_vessel' reagent='water'/>
          <Separate purpose='extract' product_phase='top' from_vessel='DMF_vessel' separation_vessel='ether_vessel' to_vessel='product_vessel'/>
          <RunColumn from_vessel='product_vessel' to_vessel='product_vessel' column='silica gel'/>
      </Procedure>
  </Synthesis>
</XDL>
```

**Comment by fuyun-cas** [comment]
Yes, It works well with the Google Colab. Thanks for your reply.

**Comment by fuyun-cas** [comment]
hi, another question: XDL_description.txt only 7 tags have ditailed information. why not describe the other tags, such as EvacuateAndRefill, Purge.

**Comment by n-yoshikawa** [comment]
Thank you for your question. Since the token length of LLM is limited, we provided detailed descriptions only for frequently used tags. The verifier provides additional descriptions when the generated details are incorrect.
